File: Maths/Collatz.py
def collatz(n:int):
    while(n!=1):
        print(n, "->", end=" ")
        if n%2!=0:
            n = 3*n+1
        else:
            n = n//2
    print(n)

if __name__ =="__main__":
    n = int(input())
    collatz(n)

File: Maths/Collatz.cpp
/*
Implementation of Collatz conjecture.
*/

#include <bits/stdc++.h>
using namespace std;

class Solution {
    public:
    static void collatz (int n) {
        while(n!=1) {
            cout << n << "->";
            if(n&1) {
                n = 3*n+1;
            }
            else {
                n = n/2;
            }
        }
        cout << n;
    }
};

int main () {
    int n;
    cin >> n;
    Solution::collatz(n);
    return 0;
}